### PR TITLE
[Improve][connector-v2]optimized MySQL&SqlServer cdc connector

### LINKED ISSUE #5349: [Feature][CDC] Improve snapshot read performance
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

When closing ExactlyOnce, the memory buffer can be canceled and read data should be sent to the sink immediately.

reference
https://github.com/apache/seatunnel/blob/dev/seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/base/source/reader/external/IncrementalSourceScanFetcher.java#L123

https://github.com/apache/seatunnel/blob/dev/seatunnel-connectors-v2/connector-cdc/connector-cdc-mysql/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/source/reader/fetch/scan/MySqlSnapshotSplitReadTask.java#L235

image

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by sunxiaojian**
## Purpose of this pull request

close #5349

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by sunxiaojian** [comment]
@hailin0 PTAL

File: seatunnel-connectors-v2/connector-cdc/connector-cdc-mysql/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/source/reader/fetch/scan/MySqlSnapshotSplitReadTask.java [hunk 1]
```diff
@@ -169,7 +170,8 @@ private void createDataEvents(
             TableId tableId)
             throws Exception {
         EventDispatcher.SnapshotReceiver snapshotReceiver =
-                dispatcher.getSnapshotChangeEventReceiver();
+                dispatcher.getIncrementalSnapshotChangeEventReceiver(
+                        StreamingChangeEventSourceMetrics.NO_OP);
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-cdc/connector-cdc-mysql/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/source/reader/fetch/scan/MySqlSnapshotSplitReadTask.java]:
```suggestion
                exactlyOnce ? dispatcher.getSnapshotChangeEventReceiver() : dispatcher.getIncrementalSnapshotChangeEventReceiver(
                        StreamingChangeEventSourceMetrics.NO_OP);
```

**Review by hailin0 — COMMENTED** [review]
Please update the sqlserver-cdc module in association

**Comment by sunxiaojian** [comment]
Before optimization：

env {
  # You can set engine configuration here
  execution.parallelism = 1
  job.mode = "STREAMING"
  checkpoint.interval = 5000
}

source {
  MySQL-CDC {
     result_table_name = "customers_mysql_cdc"
    server-id = 5660
    username = “*****”
    password = “******”
    table-names = [“test_database.source_table"]
    exactly_once = false
    base-url = "jdbc:mysql://localhost:3306/test_database?useSSL=false"
    catalog {
      factory = MySQL
    }
  }
}

sink {
  Console {
  }
}

image



After optimization：

We can configure the data size for each batch processing through 'debezium.max.batch.size’, 

env {
  # You can set engine configuration here
  execution.parallelism = 1
  job.mode = "STREAMING"
  checkpoint.interval = 5000
}

source {
  MySQL-CDC {
    result_table_name = "customers_mysql_cdc"
    server-id = 5660
    debezium.max.batch.size = 300
    username = “*****”
    password = “******”
    table-names = [“test_database.source_table"]
    exactly_once = false
    base-url = "jdbc:mysql://localhost:3306/test_database?useSSL=false"
    catalog {
      factory = MySQL
    }
  }
}

sink {
  Console {
  }

}

image

**Pull request closed without merging**